### DISCUSSION TITLE
Can SQLModel be more generic in its implementations? Swappable backends?

### DISCUSSION THREAD

**Discussion by ashleysommer** [fastapi/sqlmodel#1923] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Description

SQLModel has emerged as the long-awaited missing link that uses type annotations to build generic data models to bridge the gap between data validators (like Pydantic) and ORM layer (like SQLAlchemy).

However at the moment SQLModel is tied directly to Pydantic and SQLAlchemy, and works only with Pydantic and SQLAlchemy.

I wonder if SQLModel can be a more generic bridge, to be used between _any_ modern data validator, and ORM.

### Wanted Solution

I'd like a stand-alone version of SQLModel that can be installed without requiring Pydantic and SQLModel to be installed along with it.

### Operating System

Linux

### Operating System Details

N/A

### SQLModel Version

0.0.4

### Python Version

3.8

**Comment by jykae** [comment]
Also interested in this, would like to see @jordaneremieff 's [Djantic](https://github.com/jordaneremieff/djantic) story to continue. Django has a great ORM, and would like it to play well with Pydantic as well.

**Comment by YuriiMotov** [comment, marked as answer]
I think this is beyond the scope of this project.
Maintaining the support of multiple backends is very heavy burden and it's reasonable to choose the most popular ones and only support them
